EternalBAND.Win/Controllers/WebApi/AuthentiationWebController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using EternalBAND.Api.Options;
+using EternalBAND.DomainObjects;
+using EternalBAND.DomainObjects.ApiContract;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace EternalBAND.Win.Controllers.WebApi
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class AuthenticationWebController : ControllerBase
+    {
+        private readonly SignInManager<Users> signInManager;
+        private readonly JwtTokenOptions jwtTokenSettings;
+        private readonly ILogger<AuthenticationWebController> logger;
+        private readonly UserManager<Users> userManager;
+
+        // TODO create an AuthenticationService to separate logic
+        public AuthenticationWebController(
+            ILogger<AuthenticationWebController> logger, 
+            SignInManager<Users> signInManager, 
+            IOptions<JwtTokenOptions> jwtTokenSettings,
+            UserManager<Users> userManager
+            )
+        {
+            this.logger = logger;
+            this.signInManager = signInManager;
+            this.jwtTokenSettings = jwtTokenSettings.Value;
+            this.userManager = userManager;
+        }
+        // for now UserName and email is same later we should separate them
+        [HttpPost("login")]
+        public async Task<IActionResult> Login([FromBody] UserContracts loginUser)
+        {
+            var result = await signInManager.PasswordSignInAsync(loginUser.Username, loginUser.Password, loginUser.RememberMe, lockoutOnFailure: false);
+            if (!result.Succeeded)
+            {
+                if (result.IsNotAllowed)
+                {
+                    return Unauthorized("User is not allowed");
+                }
+
+                else if (result.IsLockedOut)
+                {
+                    return Unauthorized("User account is locked");
+                }
+
+                else if (result.RequiresTwoFactor)
+                {
+                    return Unauthorized("Need 2 FActor Authentication");
+                }
+
+                else
+                {
+                    return Unauthorized("Username or password is incorrect");
+                }
+            }
```

Comment:
if (!result.Succeeded)
            {
                if (result.IsNotAllowed)
                {
                    return Unauthorized("User is not allowed");
                }

                else if (result.IsLockedOut)
                {
                    return Unauthorized("User account is locked");
                }

                else if (result.RequiresTwoFactor)
                {
                    return Unauthorized("Need 2 FActor Authentication");
                }

                else
                {
                    return Unauthorized("Username or password is incorrect");
                }
            }
            
            
I feel it these lines is written the late nights. Lets discuss it just before we enter Pubg.

Reply:
ControllerBase Class has lots of helpful method but I did not discover it honestly.
What we can do is creating dedicated controller or middleware that handle status codes ( We can change in ErrorWebController or convert it to a middleware )

Reply:
@erenkorkmazemre  added an extension method HandleSignInFailure to cover signIn problems please take a look I believe it is clearer now